### PR TITLE
fixes #25255 - show error when host cannot be deleted

### PR DESCRIPTION
**Pull request by timogoebel**

### CODE REVIEW

**Comment by timogoebel** [comment]
[test katello]

**Comment by timogoebel** [comment]
[test katello]

**Comment by jlsherrill** [comment]
Change looks good off the bat, but a broken test

File: test/controllers/foreman/hosts_controller_test.rb [hunk 1]
```diff
@@ -66,4 +66,28 @@ def test_csv_export_search
...
+
+    test 'should destroy a host' do
+      delete :destroy, params: { :id => host.name }
+      assert_nil flash[:error]
+      assert_not_nil flash[:success]
+      assert_redirected_to hosts_url
+    end
+
+    test 'shows an error when host can not be destroyed' do
+      ::Katello::RegistrationManager.any_instance.stubs(:unregister_host).returns(false)
```
Comment (akofink) [inline test/controllers/foreman/hosts_controller_test.rb]: It's a class method, not an instance method. Removing `.any_instance` seems to fix the test.

```rb
::Katello::RegistrationManager.stubs(:unregister_host).returns(false)
```

**Comment by timogoebel** [comment]
Tests are all green now.

**Comment by timogoebel** [comment]
@akofink, @jlsherrill: Is there anything more todo here?

**Comment by akofink** [comment]
Just to merge it!

**Comment by timogoebel** [comment]
@akofink: Thanks, can we get this into 3.8.1?

**Comment by akofink** [comment]
@sjha4 Are you ok putting this in 3.8.1?

**Comment by sjha4** [comment]
Looks like a good change to get into 3.8.1 :+1:

**Pull request merged**